src/gui/tray/activitydata.cpp

```diff
@@ -93,8 +113,12 @@ OCC::Activity Activity::fromActivityJson(const QJsonObject &json, const AccountP
                 parameterJsonObject.value(QStringLiteral("id")).toString(),
                 parameterJsonObject.value(QStringLiteral("name")).toString(),
                 parameterJsonObject.contains(QStringLiteral("path")) ? parameterJsonObject.value(QStringLiteral("path")).toString() : QString(),
-                parameterJsonObject.contains(QStringLiteral("link")) ? QUrl(parameterJsonObject.value(QStringLiteral("link")).toString()) : QUrl(),
+                stringToUrl(account->url(), parameterJsonObject.value(QStringLiteral("link")).toString()),
             };
+
+            if (activity._objectType == QStringLiteral("calendar") && activity._link.isEmpty()) {
+                activity._link = activity._subjectRichParameters[i.key()].link;
+            }
```

Comment:
nitpick: It would be safer/faster/cleaner to declare the link used here outside of the `Activity::RichSubjectParameter` and use it in both the `Activity::RichSubjectParameter` and the calendar-related if statement